integration-tests/tests/simulation.py

```diff
@@ -0,0 +1,88 @@
+from pprint import pprint
+import abc
+
+import json
+
+class BaseSnapshotProcessor():
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def receive_snapshot(self, information):
+        pass
+
+class SnapshotProcessor(BaseSnapshotProcessor):
+    def __init__(self, binder, player_id = 1):
+        self.binder = binder
+        self.world_states = []
+        self.player_id = player_id
+        self.player_added = False
+
+    def check_player_moved_at_most_one_space(self, last_pos, curr_pos):
+        if last_pos is None:
+            return
+        if curr_pos is None:
+            return
+        x_diff = last_pos['x'] - curr_pos['x']
+        y_diff = last_pos['y'] - curr_pos['y']
+
+        x_move = x_diff >= -1 and x_diff <= 1 and y_diff == 0
+        y_move = y_diff >= -1 and y_diff <= 1 and x_diff == 0
+
+        print("Checking player movement...")
+        print(str((last_pos['x'], last_pos['y'])) + " -> " + str((curr_pos['x'], curr_pos['y'])))
+        if x_diff == 0 and y_diff == 0:
+            print("No movement > OK")
+        elif x_move:
+            print("Movement on x axes > OK")
+        elif y_move:
+            print("Movement on y axes > OK")
+        else:
+            print("To many moves > FAIL")
```

Comment:
Too many, not to many :) However I wouldn't recommend prints in tests - use asserts please

Reply:
:laughing: I'm awful, sorry.